plugins/star-rating/frontend/public/javascripts/countly.models.js

```diff
@@ -87,6 +92,151 @@
+                trigger_button_text: feedbackWidget.trigger_button_text,
+                target_devices: JSON.stringify(feedbackWidget.target_devices),
+                target_page: feedbackWidget.target_page,
+                target_pages: JSON.stringify(feedbackWidget.target_pages),
+                is_active: feedbackWidget.is_active,
+                app_key: countlyGlobal['apps'][store.get("countly_active_app")].key,
+            },
+            success: function (json, textStatus, xhr) {
+                callback(json, xhr.status);
+            }
+        })
+    }
+
+    starRatingPlugin.editFeedbackWidget = function(feedbackWidget, callback) {
+        return $.ajax({
+            type: "GET",
+            url: countlyCommon.API_URL + "/i/web-feedback/widgets/edit",
+            data: {
+                popup_header_text: feedbackWidget.popup_header_text,
+                popup_comment_callout: feedbackWidget.popup_comment_callout,
+                popup_email_callout: feedbackWidget.popup_email_callout,
+                popup_button_callout: feedbackWidget.popup_button_callout,
+                popup_thanks_message: feedbackWidget.popup_thanks_message,
+                trigger_position: feedbackWidget.trigger_position,
+                trigger_bg_color: feedbackWidget.trigger_bg_color,
+                trigger_font_color: feedbackWidget.trigger_font_color,
+                trigger_button_text: feedbackWidget.trigger_button_text,
+                target_devices: feedbackWidget.target_devices,
+                target_page: feedbackWidget.target_page,
+                target_pages: feedbackWidget.target_pages,
+                is_active: feedbackWidget.is_active,
+                app_key: countlyGlobal['apps'][store.get("countly_active_app")].key,
+                widget_id: feedbackWidget._id
+            },
+            success: function (json, textStatus, xhr) {
+                callback(json, xhr.status);
+            }
+        })
+    }
+
+    starRatingPlugin.removeFeedbackWidget = function(widget_id, with_data, callback) {
+        return $.ajax({
+            type: "GET",
+            url: countlyCommon.API_URL + "/i/web-feedback/widgets/remove",
+            data: {
+                app_key: countlyGlobal['apps'][store.get("countly_active_app")].key,
+                widget_id: widget_id,
+                with_data: with_data
+            },
+            success: function (json, textStatus, xhr) {
+                callback(json, xhr.status);
+            }
+        })
+    }
+
+    /**
+     * This is for fetching feedback comments objects from server side 
+     * @namespace starRatingPlugin
+     * @method requestFeedbackData
+     * @param {}
+     * @return {func} ajax func to request data and store in _fd
+     */
+    starRatingPlugin.requestFeedbackWidgetsData = function() {
+        var period = countlyCommon.getPeriod();
+        var periodString = typeof period === "object" ? "[" + period.toString() + "]" : period;
```

Comment:
There is a method to get period for ajax directly, without the need of special case processing:
https://code.count.ly/server/browser/countlyCommon.html#.getPeriodForAjax